tests/testsuite/build_script.rs

```diff
@@ -6087,3 +6087,40 @@ fn directory_with_leading_underscore() {
         .with_status(0)
         .run();
 }
+
+#[cargo_test]
+fn linker_search_path_preference() {
```

Comment:
This looks nice. Thank you!

Could we have one more test that: Has some dependencies, and those dependencies also emit `cargo::rustc-link-search`, and we ensure that the order from the dependencies is maintained?

Reply:
I just added dependencies into the test rather than creating a separate test.